src/class.c

```diff
@@ -1643,13 +1657,22 @@ bool InterfaceDeclaration::isBaseOf(ClassDeclaration *cd, int *poffset)
                 *poffset = b->offset;
                 if (j && cd->isInterfaceDeclaration())
                     *poffset = OFFSET_RUNTIME;
+
+                /* TODO: Even though it's an interface to base interface upcast,
+                 * I think we can avoid runtime offset determination ultimately.
+                 * (I doubt that it was just a workaround for the bug in the
+                 * inferface to Object downcast)
+                 */
```

Comment:
@rainers I noticed that we may use static cast always for any upcasts. I don't change it in this PR because it would introduce some runtime behavior change, but it would be a possible "optimization".